compiler/pipes/final-check.cpp

```diff
@@ -587,3 +584,37 @@ void FinalCheckPass::raise_error_using_Unknown_type(VertexPtr v) {
     kphp_error(0, "Using Unknown type");
   }
 }
+
+void FinalCheckPass::check_magic_methods(FunctionPtr fun) {
+  if (!fun->modifiers.is_instance()) {
+    return;
+  }
+
+  const auto name = fun->local_name();
+
+  if (name == ClassData::NAME_OF_TO_STRING) {
+    check_magic_tostring_method(fun);
+  } else if (name == ClassData::NAME_OF_CLONE) {
+    check_magic_clone_method(fun);
+  }
+}
+
+void FinalCheckPass::check_magic_tostring_method(FunctionPtr fun) {
+  stage::set_function(fun);
```

Comment:
No need to do this. `set_function()` for `current_function` performs implicitly.

Reply:
Fixed.